cmd/snap/error.go

```diff
@@ -100,6 +100,12 @@ func errorToCmdMessage(snapName string, e error, opts *client.SnapOptions) (stri
 		// arch/channel/revision. Surface that here somehow!
 
 		msg = i18n.G("snap %q not found")
+		if snapName == "" {
```

Comment:
does it matter if we always believe the server instead? basically if Value != "" use that as snapName, also given that it can be "" it's a bit strange that snapName is the first argument to the function to me